apps/ledger-live-mobile/src/families/evm/EvmFeesStrategy.tsx

```diff
@@ -74,6 +81,32 @@ export default function EvmFeesStrategy({
   const [customStrategyTransactionPatch, setCustomStrategyTransactionPatch] =
     useState<Partial<Transaction>>();
 
+  useEffect(() => {
+    if (!customStrategyTransactionPatch) {
+      return;
+    }
+    const bridge = getAccountBridge<Transaction>(account, parentAccount);
+
+    /**
+     * If the customStrategyTransactionPatch is present, this means the custom
+     * fee has been edited by the user. In this case, we need to update the
+     * transaction with the new fee data.
+     * This also selects the new "custom" strategy in the fees strategy list.
+     */
+    const updatedTransaction = bridge.updateTransaction(transaction, {
+      ...customStrategyTransactionPatch,
+    });
+
+    setTransaction(updatedTransaction);
```

Comment:
how can this code ever converge? this will make `transaction` being a new ref so the effect will trigger again, as soon as customStrategyTransactionPatch is set